homeassistant/components/fyta/sensor.py

```diff
@@ -0,0 +1,171 @@
+    ),
+    FytaSensorEntityDescription(
+        key="moisture_status",
+        translation_key="fyta_moisture_status",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:water-percent-alert",
+    ),
+    FytaSensorEntityDescription(
+        key="salinity_status",
+        translation_key="fyta_salinity_status",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:sprout-outline",
+    ),
+    FytaSensorEntityDescription(
+        key="temperature",
+        translation_key="fyta_temperature",
+        native_unit_of_measurement="°C",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:thermometer",
+    ),
+    FytaSensorEntityDescription(
+        key="light",
+        translation_key="fyta_light",
+        native_unit_of_measurement="mol/d",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:weather-sunny",
+    ),
+    FytaSensorEntityDescription(
+        key="moisture",
+        translation_key="fyta_moisture",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.MOISTURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:water-percent",
+    ),
+    FytaSensorEntityDescription(
+        key="salinity",
+        translation_key="fyta_salinity",
+        native_unit_of_measurement="mS/cm",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:sprout-outline",
+    ),
+    FytaSensorEntityDescription(
+        key="ph",
+        translation_key="fyta_PH",
+        device_class=SensorDeviceClass.PH,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:ph",
+    ),
+    FytaSensorEntityDescription(
+        key="battery_level",
+        translation_key="fyta_battery_level",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        icon="mdi:battery",
+    ),
+    FytaSensorEntityDescription(
+        key="last_updated",
+        translation_key="fyta_last_updated",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        icon="mdi:update",
+    ),
+    FytaSensorEntityDescription(
+        key="plant_number",
+        translation_key="fyta_plant_number",
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        icon="mdi:update",
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the FYTA binary sensors."""
+    coordinator: FytaCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [
+            FytaSensor(coordinator, entry, sensor)
+            for sensor in SENSORS
+            if sensor.key in coordinator.data
+        ]
+    )
+
```

Comment:
So wait, the controller has the same sensors as a plant?

Reply:
A few are the same.

Reply:
But do we create all of them for both? Do both support it?

Reply:
Oh I now see the if statement. I think that's okay